### ISSUE TITLE
Google Captcha Problems with required / Form submit in storefront

### ISSUE THREAD

**Issue by LeonIllenseer** [shopware/platform#6456]
*PHP Version:* 8.3   | *Shopware Version:* ^6.6  | *Affected area / extension:* Platform(Default) 

 ----

*_Actual behaviour:_*

Google Captcha v2 / v3 throw Error Message:

"An invalid form control with name='_grecaptcha_v2' is not focusable"

The Issue can be fixed by changing the Input to type="hidden" in v2 and v3

../src/Storefront/Resources/views/storefront/component/captcha/googleReCaptchaV2.html.twig
Core code: 

{% block component_captcha_google_re_captcha_v2 %}
    {% set googleReCaptchaV2Options = {
        siteKey: config('core.basicInformation.activeCaptchasV2.googleReCaptchaV2.config.siteKey'),
        invisible: config('core.basicInformation.activeCaptchasV2.googleReCaptchaV2.config.invisible')
    } %}

    <div class="captcha-google-re-captcha-v2"
         data-google-re-captcha-v2="true"
         data-google-re-captcha-v2-options="{{ googleReCaptchaV2Options|json_encode }}">
        <input
            type="text"
            class="d-none grecaptcha-v2-input"
            name="{{ constant('Shopware\\Storefront\\Framework\\Captcha\\GoogleReCaptchaV2::CAPTCHA_REQUEST_PARAMETER') }}"
            data-skip-report-validity="true"
            required>

        <div class="grecaptcha-v2-container"></div>

        {% if googleReCaptchaV2Options.invisible %}
            {% set recaptchaV3Active = config('core.basicInformation.activeCaptchasV2.googleReCaptchaV3.isActive') %}

            {% if not recaptchaV3Active %}
                <div class="data-protection-information grecaptcha-protection-information">
                    {{ 'captcha.googleReCaptcha.dataProtectionInformation'|trans|sw_sanitize }}
                </div>
            {% endif %}
        {% endif %}
    </div>
{% endblock %}

*_Expected behaviour:_*

Normal Form submit without errors

*_How to reproduce:_*

Select Google Captcha v2 or v3 make configurations and try to submit a form in storefront.

**Comment by shopwareBot** [comment]
[public] Shopware automated response: This issue is linked to the internal issue: **NEXT-40458**



[created from NEXT-40458, comment 536424]

**Comment by shopwareBot** [comment]
[public] 

 > Priority changed to: **Low**

 Thank you for your report. After careful consideration, we've concluded that this issue does not require immediate attention or isn't something we can address right now. We prioritize openness and clear communication with our community, so we want to be transparent. We may not resolve all reported issues, and this one might remain unresolved. To stay effective, we periodically clean our backlog, and this issue could be closed in the future.

 For now, we'll keep this ticket open to invite further feedback from the community in case we've underestimated its impact and to allow us to revisit it later.



[created from NEXT-40458, comment 536702]
